tracing-journald/src/lib.rs

```diff
@@ -353,6 +437,36 @@ fn put_priority(buf: &mut Vec<u8>, meta: &Metadata) {
     );
 }
 
+fn put_facility(buf: &mut Vec<u8>, facility: SyslogFacility) {
+    let value = match facility {
+        SyslogFacility::Auth => libc::LOG_AUTH,
+        SyslogFacility::Authpriv => libc::LOG_AUTHPRIV,
+        SyslogFacility::Cron => libc::LOG_CRON,
+        SyslogFacility::Daemon => libc::LOG_DAEMON,
+        SyslogFacility::Ftp => libc::LOG_FTP,
+        SyslogFacility::Kern => libc::LOG_KERN,
+        SyslogFacility::Local0 => libc::LOG_LOCAL0,
+        SyslogFacility::Local1 => libc::LOG_LOCAL1,
+        SyslogFacility::Local2 => libc::LOG_LOCAL2,
+        SyslogFacility::Local3 => libc::LOG_LOCAL3,
+        SyslogFacility::Local4 => libc::LOG_LOCAL4,
+        SyslogFacility::Local5 => libc::LOG_LOCAL5,
+        SyslogFacility::Local6 => libc::LOG_LOCAL6,
+        SyslogFacility::Local7 => libc::LOG_LOCAL7,
+        SyslogFacility::Lpr => libc::LOG_LPR,
+        SyslogFacility::Mail => libc::LOG_MAIL,
+        SyslogFacility::News => libc::LOG_NEWS,
+        SyslogFacility::Syslog => libc::LOG_SYSLOG,
+        SyslogFacility::User => libc::LOG_USER,
+        SyslogFacility::Uucp => libc::LOG_UUCP,
+    };
+    put_field_wellformed(
+        buf,
+        "SYSLOG_FACILITY",
+        format!("{}", value >> 3).as_bytes(),
```

Comment:
it's unfortunate that the use of `format!` here allocates when the data is about to be written to the internal buffer. we should try to avoid allocations like this on every log call, and prefer formatting data directly into the buffer. i think it would actually be preferable to use `put_field_length_encoded` here: https://github.com/tokio-rs/tracing/blob/d30785606dae2f609b9d2ee4576ebf7f047773da/tracing-journald/src/lib.rs#L499
and use `write!(buf, value >> 3)` in the closure, so that we don't do the intermediate allocation?

@Ralith am I correct that this would be preferable? or is there another way we ought to do it?

Reply:
Because the key and value are both known to be well-formed, this can be done more simply with e.g. `writeln!(buf, "SYSLOG_FACILITY={}", value)`.